tune_sklearn/_trainable.py

```diff
@@ -95,6 +95,10 @@ def _setup(self, config):
                 # max_iter is to make sklearn only fit for one epoch,
                 # while max_iters (which the user can set) is the usual max
                 # number of calls to _trainable.
+                if "max_iter" in self.estimator_config:
+                    raise ValueError("tune-sklearn uses `max_iter` to warm "
+                                     "start, so this parameter can't be "
+                                     "set when warm start early stopping. ")
```

Comment:
Instead of `_trainable` do you think we can do this in `tune_basesearch.__init__`?

This is so that we can raise an error faster (Before starting Ray).

Reply:
We'd have to do it in the subclasses because the parameter dict that we have to check aren't passed to the superclass. I thought it would be nicer to do it when they call fit so we don't have to have duplicate code in the subclasses.

Reply:
ah good point - i guess my main concern is that the ray tune stacktrace is not as nice as if you just raised it in the local class (there's like a bunch of raylet stuff and pyx stuf). You could also define a helper method that encapsulates the ValueError logic right?